scripts/install/auto-install.sh

```diff
@@ -0,0 +1,245 @@
+"
+  time do_sudo chown --recursive "${user}:${group}" "${path}"
+  echo
+}
+
+function ensure_directory_ownership() {
+  local -r path="${1}"
+  local -r user="${2}"
+  local -r group="${3}"
+
+  echo_bar
+  echo "
+Ensuring that directory ${path} exists.
+"
+  if [ ! -d "${path}" ]
+  then
+    do_sudo mkdir -p "${path}"
+    echo
+  fi
+
+  ensure_ownership "${path}" "${user}" "${group}"
+}
+
+# End of functions.
+################################################################################
+
+# Exit immediately if a command fails:
+set -e
+
+# COPIED from default-env-vars.sh
+[[ -z "${PANUSER}" ]] && export PANUSER="panoptes"      # Default user
+[[ -z "${PANDIR}" ]] && export PANDIR="/var/panoptes"   # Main Dir
+[[ -z "${PANLOG}" ]] && export PANLOG="${PANDIR}/logs"  # Log files
+[[ -z "${POCS}" ]] && export POCS="${PANDIR}/POCS"      # Main Observatory Control
+[[ -z "${PAWS}" ]] && export PAWS="${PANDIR}/PAWS"      # Web Interface
+
+# Do we need to create the user PANUSER?
+
+if ! id -u "${PANUSER}" 2>/dev/null 1>/dev/null
+then
+  >&2 cat <<ENDOFMESSAGE
+The user ${PANUSER} doesn't exist yet. Please create it by running:
+
+  sudo adduser --shell /bin/bash --add_extra_groups ${PANUSER}
+
+You will be prompted to enter a password for this new user. You may also be
+prompted to provide contact info as if this were a user on a shared computer
+(e.g. Full Name and Home Phone). Press Enter to leave these unset.
+
+After the command completes, we need to ensure that the user is a
+member of some key groups:
+
+  sudo usermod --append --groups adm,dialout,plugdev,sudo ${PANUSER}
+
+Next, login as ${PANUSER} and re-execute the command you used to
+run this script.
+ENDOFMESSAGE
+
+  maybe_print_example
+  exit 1
+fi
+
+# Do we need to login as the user PANUSER?
+
+if [[ "$(whoami)" != "${PANUSER}" ]]
+then
+  echo >&2 "
+This script should be executed by the user ${PANUSER}, not as $(whoami).
+Please login as ${PANUSER} and re-execute the command you used to
+run this script.
+"
+
+  maybe_print_example
+  exit 1
+fi
+
+# I (James Synge) have noticed that if $HOME/.cache/ doesn't exist, then at some
+# point during the install it gets created, but is owned by root:root, which
+# then messes up later steps that attempt to use it. So, we make sure here that
+# it exists with the correct ownership.
+
+PANGROUP="$(id -gn "${PANUSER}")"
+ensure_directory_ownership "${HOME}/.cache" "${PANUSER}" "${PANGROUP}"
+
+# Do the same with PANDIR (usually /var/panoptes).
+ensure_directory_ownership "${PANDIR}" "${PANUSER}" "${PANGROUP}"
+
+if [ ! -x "$(safe_which git)" ]
+then
+  echo_bar
+  echo "
+git is not installed. Updating package cache, then installing git.
+"
+  do_sudo apt-get update
+  echo
+  echo
+  do_sudo apt-get install -y git
+  echo
+fi
+
+################################################################################
+# Clone the POCS repo from github.
+
+clone_or_update "${POCS}" https://github.com/panoptes/POCS.git upstream develop
```

Comment:
Yes. This is used as the origin name when cloning, so that `https://github.com/panoptes/POCS.git` will be the upstream repo, and `https://github.com/${POCS_GITHUB_USER}/POCS.git` will be the origin.